middleware/timeout.go

```diff
@@ -0,0 +1,31 @@
+package middleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-martini/martini"
+)
+
+type TimeoutContext interface {
+	context.Context
+	MartiniContext() martini.Context
+}
+
+type timeoutContext struct {
+	context.Context
+	martiniContext martini.Context
+}
+
+func (t *timeoutContext) MartiniContext() martini.Context {
+	return t.martiniContext
+}
+
+// WithFiveMinuteTimout adds a five minute timeout context to the request.
+// it explicitly ignores the context of the http request.
```

Comment:
the context of the http request would come from
```go
r.Context()
```
where r is a `*http.Request`. 
That context gets canceled when the socket closes from the client's side. We are ignoring that fact because we don't want to stop pulling the image.